File: src/main/kotlin/org/psota/invoices/entity/BankInfo.java
package org.psota.invoices.entity;

import jakarta.persistence.Entity;
import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

@ToString(callSuper = true)
@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
@SuperBuilder(toBuilder = true)
@NoArgsConstructor(access = AccessLevel.PROTECTED) // for JPA
@Getter
@Entity
public class BankInfo extends BaseEntity {

  private String accountNo;
  private String iban;
  private String bankName;
  private String swiftCode;
  private String paymentMethod;

}

File: invoices-be/src/main/kotlin/org/psota/invoices/entity/BankInfo.kt
package org.psota.invoices.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity

@Entity
class BankInfo(
        @Column(nullable = false)
        val accountNo: String,
        @Column(nullable = false)
        val iban: String,
        @Column(nullable = false)
        val bankName: String,
        @Column(nullable = false)
        val swiftCode: String,
        @Column(nullable = true)
        val paymentMethod: String
) : BaseEntity()
